daft/expressions/expressions.py

```diff
@@ -1011,6 +1011,18 @@ def right(self, nchars: int | Expression) -> Expression:
         nchars_expr = Expression._to_expression(nchars)
         return Expression._from_pyexpr(self._expr.utf8_right(nchars_expr._expr))
 
+    def find(self, substr: str | Expression) -> Expression:
+        """Returns the index of the first occurrence of the substring in each string
```

Comment:
We should also document that this will return `-1` if the string is not found, and that all results returned are 0-indexed.

Noting here also that this is a departure from:

1. Spark/PostgreSQL behavior (returns 0 if not found, but the results it returns are 1-indexed)
2. Polars behavior (return null if not found)

This -1 behavior is consistent with Python, Pandas and Ibis behavior